Runtime/Behaviours/CinemachineConfiner.cs

```diff
@@ -100,15 +101,36 @@ public bool IsValid
             }
         }
 
+        protected override void ConnectToVcam(bool connect)
+        {
+            base.ConnectToVcam(connect);
+
+            CinemachineVirtualCamera vcam = VirtualCamera as CinemachineVirtualCamera;
+            var components = vcam.GetComponentPipeline();
+            foreach (var component in components)
+            {
+                if (component.BodyAppliesAfterAim)
+                {
+                    var extraState = GetExtraState<VcamExtraState>(VirtualCamera);
+                    extraState.isFramingTransposer = true;
+                    break;
+                }
+            }
+        }
+        
+
         /// <summary>Callback to to the camera confining</summary>
         protected override void PostPipelineStageCallback(
             CinemachineVirtualCameraBase vcam,
             CinemachineCore.Stage stage, ref CameraState state, float deltaTime)
         {
             if (IsValid)
             {
-                // Move the body before the Aim is calculated
-                if (stage == CinemachineCore.Stage.Body)
+                var extraState = GetExtraState<VcamExtraState>(VirtualCamera);
+                if (extraState.isFramingTransposer && stage == CinemachineCore.Stage.Finalize
```

Comment:
do you need brackets here?

Reply:
also: you need to use vcam, not VirtualCamera as parameter, to get the right extraState.
It's done just below - you should use that, no need to get extraState twice.

Reply:
ok, you don't *need* brackets, but you should put them, because nobody remembers the order of evaluation of && and || (at least I don't :) )

Reply:
No brackets are needed, however I agree it is more clear if I put brackets.